pidfile/pidfile.go

```diff
@@ -0,0 +1,30 @@
+package pidfile
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+// WritePid : write pid to given file
+func WritePid(pidfile string) error {
+	dir, filename := filepath.Split(pidfile)
+	tmpfile, err := ioutil.TempFile(dir, filename+".*")
+	if err != nil {
+		return errors.Wrap(err, "Cloud not create tempfile")
+	}
+	_, err = tmpfile.WriteString(fmt.Sprintf("%d", os.Getpid()))
+	if err != nil {
+		tmpfile.Close()
```

Comment:
In this case, `tmpfile` remains. Removing file is caller's responsibility.

refs -> https://golang.org/pkg/io/ioutil/#TempFile